### PR TITLE
update the build

### PR DESCRIPTION
**Pull request by johnynek**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #sbt
 /project/target/
 /project/project/target
-
+sonatype.sbt
```
Comment (ianoc) [inline .gitignore]: does one normally not put this into the ~/.sbt/0.13 folder ?

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,307 @@
+import com.typesafe.tools.mima.plugin.MimaPlugin.mimaDefaultSettings
```
Comment (ianoc) [inline build.sbt]: Why move from project/Build.scala to a build.sbt? is this now the more blessed way?

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 )
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-ghpages" % "0.5.2")
-
-addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1.6")
-
-addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.7.0")
-
-addSbtPlugin("com.typesafe.sbt" % "sbt-scalariform" % "1.3.0")
+addSbtPlugin("com.github.gseitz" % "sbt-release"     % "1.0.0")
```
Comment (ianoc) [inline project/plugins.sbt]: where does this formatting style come from ?

Comment (johnynek) [inline project/plugins.sbt]: me personally... :) I just cleaned it when adding do it.

Comment (ianoc) [inline project/plugins.sbt]: Could we use scalriform? the old one is closer to how we format most of our scala I suppose

File: build.sbt (hunk 2 above)
Comment (johnynek) [inline build.sbt]: yes, and the plugins are a pain to config without it. There is some autoImport thing that is now standard, so people assume you are using that.

File: .gitignore (hunk 1 above)
Comment (johnynek) [inline .gitignore]: well, sbt release has some possibly per project options, so I put it here, but it won't publish a dirty directory.

File: project/plugins.sbt (hunk 3 above)
Comment (johnynek) [inline project/plugins.sbt]: Happy to. Can it run on .sbt files?

**Pull request merged**